### PR TITLE
[ubuntu-dev] dependency greenkeeping & small improvements

### PR DESCRIPTION
**Pull request by jankeromnes**

### CODE REVIEW

**Comment by jankeromnes** [comment]
@bnjbvr Could you please have a quick look over this whenever you have some time? 😄

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 1]
```diff
@@ -117,17 +123,21 @@ RUN git clone https://github.com/ninja-build/ninja /tmp/ninja \
  && git checkout v1.8.2 \
  && ./configure.py --bootstrap \
  && sudo mv ninja /usr/bin/ninja \
+ && mv misc/bash-completion /home/user/.ninja-bash-completion \
+ && mv misc/zsh-completion /home/user/.ninja-zsh-completion \
+ && echo "\n# Ninja completion helpers." >> /home/user/.bashrc \
+ && echo ". /home/user/.ninja-bash-completion" >> /home/user/.bashrc \
```
Comment (bnjbvr) [inline ubuntu-dev/ubuntu-dev.dockerfile]: why not the same in .zshrc?

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 2]
```diff
@@ -162,8 +172,8 @@ RUN git clone https://github.com/rupa/z /home/user/.z.sh \
 # Install the latest Vim.
 RUN mkdir /tmp/vim \
  && cd /tmp/vim \
- && curl -L https://github.com/vim/vim/archive/v8.0.1071.tar.gz | tar xz \
- && cd vim-8.0.1071/src \
+ && curl -L https://github.com/vim/vim/archive/v8.0.1171.tar.gz | tar xz \
+ && cd vim-8.0.1171/src \
```
Comment (bnjbvr) [inline ubuntu-dev/ubuntu-dev.dockerfile]: Don't know if it is feasible in Dockerfiles, but wouldn't you be better off having a set of environment variables defining which version number should be downloaded, and then use these version numbers in all the curl/cd/etc lines?

File: ubuntu-dev/ubuntu-dev.dockerfile (hunk 1 above)
Comment (jankeromnes) [reply ubuntu-dev/ubuntu-dev.dockerfile]: There is no `.zshrc` file yet :/

File: ubuntu-dev/ubuntu-dev.dockerfile (hunk 2 above)
Comment (jankeromnes) [reply ubuntu-dev/ubuntu-dev.dockerfile]: Great suggestion! Thanks 😄 I'll look into using unique version variables.

**Comment by jankeromnes** [comment]
Thanks a lot! 🎆

File: ubuntu-dev/ubuntu-dev.dockerfile (hunk 1 above)
Comment (jankeromnes) [reply ubuntu-dev/ubuntu-dev.dockerfile]: I filed https://github.com/JanitorTechnology/dockerfiles/issues/96 requesting Zsh to be installed and pre-configured in all Docker images.

**Pull request merged**